src/request.ts

```diff
@@ -145,16 +145,24 @@ class Request {
     return this.execute(options);
   }
 
-  private execute<TResponse>(options: AxiosRequestConfig, resolve?: (res: TResponse) => void, reject?: (error: any) => void): Promise<TResponse> {
+  public execute<TResponse>(options: AxiosRequestConfig, resolve?: (res: TResponse) => void, reject?: (error: any) => void): Promise<TResponse> {
     if (!this._logger) {
       return Promise.reject('Logger not set on the request object');
     }
 
+    const methods = ['GET', 'POST', 'PUT', 'PATCH', 'DELETE', 'HEAD', 'OPTIONS'];
+    if (!options.method || methods.indexOf(options.method) === -1 ) {
+      return Promise.reject('No valid method set in the request options');
+    }
+
     return new Promise<TResponse>((_resolve: (res: TResponse) => void, _reject: (error: any) => void): void => {
       ((): Promise<string> => {
         if (options.headers && options.headers['x-anonymous']) {
           return Promise.resolve('');
         }
+        else if (options.headers?.authorization) {
```

Comment:
Let's avoid repeating `if (option.headers)`, instead, we can combine the first and the second clause into one and have a single return statement

Reply:
refactoring...